React v18 `npm install` package dependency compatibility

Hi there!

My project has recently upgraded to React v18 and I am currently running into a `ERESOLVE unable to resolve dependency tree` error when I try to `npm install storybook-addon-mock`. It does work if I add the `--legacy-peer-deps` flag, but ideally I'd like to avoid using it.


![image](https://user-images.githubusercontent.com/79943064/165395174-a94afd41-cd19-46ea-9360-2c94a2613c8e.png)

Any help would be greatly appreciated!

--- a/package.json
+++ b/package.json
@@ -42,8 +42,8 @@
     "react-json-editor-ajrm": "^2.5.13"
   },
   "peerDependencies": {
-    "react": "^16.8.0 || ^17.0.0",
-    "react-dom": "^16.8.0 || ^17.0.0"
+    "react":  "^16.8.0 || ^17.0.0 || ^18.0.0",
+    "react-dom":  "^16.8.0 || ^17.0.0 || ^18.0.0"
   },
   "devDependencies": {
     "@babel/cli": "^7.13.10",

